batch/batch/front_end/front_end.py

```diff
@@ -1434,18 +1466,23 @@ async def post_billing_projects_add_user(request, userdata):  # pylint: disable=
     async def insert(tx):
         row = await tx.execute_and_fetchone(
             '''
-SELECT billing_projects.name as billing_project, user
-FROM billing_projects
+SELECT billing_projects.name as billing_project,
+billing_projects.status as status,
+user FROM billing_projects
 LEFT JOIN (SELECT * FROM billing_project_users
     WHERE billing_project = %s AND user = %s FOR UPDATE) AS t
   ON billing_projects.name = t.billing_project
-WHERE billing_projects.name = %s;
+WHERE billing_projects.name = %s AND NOT billing_projects.closed;
 ''',
             (billing_project, user, billing_project))
         if row is None:
             set_message(session, f'No such billing project {billing_project}.', 'error')
             raise web.HTTPFound(deploy_config.external_url('batch', '/billing_projects'))
 
+        if row['status'] in {'closed', 'deleted'}:
```

Comment:
I think the filter already filters out closed billing projects. I don't think you need that if you're checking it here.

Reply:
yeah, good call. I actually forgot to rewrite the query from the old schema, so I fixed it and it now filters for "deleted" but not closed so it 

(I think I'll plan to treat deleted projects the same as non-existent projects everywhere except billing project creation, where you'll get an explicit error about creating a project with the same name as a deleted billing project. This means that deleted billing projects "don't exist" for basically all purposes.)